protocol/msgpack/base_iterator.go

```diff
@@ -251,6 +270,69 @@ func (it *baseIterator) decodeFloat64() float64 {
 	return value
 }
 
+func (it *baseIterator) decodeFloat64Slice(encodingType encodingType) ([]float64, pool.FloatsPool) {
+	if encodingType == nonPackedEncoding {
+		return it.decodeFloat64SliceNative()
+	}
+	return it.decodeFloat64SlicePacked()
+}
+
+func (it *baseIterator) decodeFloat64SliceNative() ([]float64, pool.FloatsPool) {
+	numValues := it.decodeArrayLen()
+	if it.decodeErr != nil {
+		return nil, nil
+	}
+	decoded, pool := it.getFloat64SliceFor(numValues)
+	for i := 0; i < numValues; i++ {
+		decoded = append(decoded, it.decodeFloat64())
```

Comment:
Seems in `decodeFloat64SlicePacked ` we check the error after decoding each value, should we do the same here?